includes/core-changes/aspnetcore/5.0/iis-urlrewrite-middleware-query-strings-are-preserved.md

```diff
@@ -0,0 +1,50 @@
+### IIS: UrlRewrite middleware query strings are preserved
+
+An IIS UrlRewrite middleware defect prevented the query string from being preserved in rewrite rules. That defect has been fixed to maintain consistency with the IIS UrlRewrite Module's behavior.
+
+For discussion, see issue [dotnet/aspnetcore#22972](https://github.com/dotnet/aspnetcore/issues/22972).
+
+#### Version introduced
+
+ASP.NET Core 5.0 Preview 7
+
+#### Old behavior
+
+Consider the following rewrite rule:
+
+```xml
+<rule name="MyRule" stopProcessing="true">
+  <match url="^foo" />
+  <action type="Redirect" url="/bar" redirectType="Temporary" appendQueryString="true" />
+</rule>
+```
+
+The preceding rule doesn't append the query string. A URI like `/foo?baz=1` redirects to `/bar` instead of `/bar?baz=1`. The `appendQueryString` attribute defaults to `true` as well.
+
+#### New behavior
+
+The query string is preserved. The URI from the example in [Old behavior](#old-behavior) would be `/foo?baz=1`.
```

Comment:
wouldn't the redirected url be `/bar?baz=1`?

Reply:
Yes. Thank you for catching that. I also had Justin update the announcement.